flocker/node/agents/blockdevice.py

```diff
@@ -1743,8 +1765,15 @@ def calculate_changes(self, configuration, cluster_state):
         local_dataset_ids = set(local_state.manifestations.keys())
 
         manifestations_to_create = set()
+        all_dataset_ids = [
+            dataset.dataset_id
+            for dataset
+            in cluster_state.all_datasets()
+        ]
```

Comment:
`list(generator expression)` avoids leaking `dataset` into the local scope which is generally a good thing.  `list(generator expression)` is probably always better than `[list comprehension]`.